activity/coordinated_worker.go

```diff
@@ -17,101 +17,105 @@ const (
 
 func (w *ActivityWorker) AddCoordinatedHandler(heartbeatInterval time.Duration, handler *CoordinatedActivityHandler) {
 
-	ticks := make(chan CoordinatedActivityHandlerTick)
-	startErr := make(chan error)
+	handlerFunc := func(ctx *ActivityContext, input interface{}) {
+		ticks := make(chan CoordinatedActivityHandlerTick)
+		startErr := make(chan error)
 
-	handlerTickFunc := func(activityTask *swf.ActivityTask, input interface{}) {
-		go func() {
-			cont, res, err := handler.Tick(activityTask, input)
-			ticks <- CoordinatedActivityHandlerTick{
-				Continue: cont,
-				Result:   res,
-				Error:    err,
-			}
-		}()
-	}
+		handlerTickFunc := func(cxt *ActivityContext, input interface{}) {
+			go func() {
+				cont, res, err := handler.Tick(cxt, input)
```

Comment:
have you thought about doing this in a `time.Ticker` loop, to rate limit it and avoid busy looping in case `handler.Tick()` is trivial (or does nothing?). This would remove the requirement for `handler.Tick()` to sleep if it terminates too quick.

Something like:

```
go func() {
	ticks := time.NewTicker(tickMinInterval)
	defer ticks.Stop()
	for {
		<-ticks
		cont, res, err := handler.Tick(ctx, input)
		// ...
	}
}
```